### ISSUE TITLE
Feature Request: Parent item attribute value of menuAttached should not cascade on childrens

### ISSUE THREAD

**Issue by skyclb** [VirtusLab-Open-Source/strapi-plugin-navigation#232]
Currently the cascading of the menu boolean on all childrens is limiting the use of `menuAttached`. In my particular use-case, i am using the navigation plugin for my breadcrumbs and menu.

Let me further explain the issue with an example:

1. Imagine you have three pages: `About us` `Team` `Vision`

2. You want to build the structure inside the plugin as the following hierarchy (to cover breadcrumb):
```
- About us
  - Team
  - Vision
```

3. Now you are not able to set `menuAttached` to `true` only for the following page: `Team`

Right now, the `menuAttached` (which is `false` in this case for the parent) gets cascade down to its children and locked on the parents value.

**Comment by ltsNotMike** [comment]
Hello @skyclb, 

Do you think a custom field of type boolean would satisfy this use case? 
If so then in version 2.1.1 there will be full support for custom fields. These fields will not propagate state down the tree so there will be no issue like you experienced with `menuAttached`.

**Comment by ltsNotMike** [comment]
Closing due to lack of response. Please feel free to reopen if needed.

**Comment by skyclb** [comment]
Hi @ltsNotMike,

Sorry, I forgot about this issue. Thanks for you reply, i think its not an issue with the field but more how the field is filled with data from the plugin itself. I took this morning an hour to dive into your code and fixed it by changing two lines, let me share with you:

1. First change is did was to remove the `data.parentAttachedToMenu` from the `disabled` condition here:
```js
// /admin/src/pages/View/components/NavigationItemForm/index.tsx#L481

// Before
disabled={!(data.isMenuAllowedLevel && data.parentAttachedToMenu)}

// After
disabled={!(data.isMenuAllowedLevel)}
```
2. Second change is did was also to remove the `data.parentAttachedToMenu` from the condition as below
```js
// /admin/src/pages/View/utils/parsers.js#L42

// Before
const itemAttachedToMenu = menuAttached && parentAttachedToMenu;

// After
const itemAttachedToMenu = menuAttached;
```

Those two changed finally resulted in allowing a child item to be `menuAttached=true` even if the parent is `false`. Look at the parent `About us` and the children `Team` in the JSON below.

I would suggest to keep both options, maybe create a plugin setting boolean named e.g "Cascade Menu Attached" which allows you to toggle the behavior.

```json

  {
    "order": 1,
    "id": 29,
    "title": "About us",
    "type": "INTERNAL",
    "path": "about-us",
    "externalPath": null,
    "uiRouterKey": "about-us",
    "menuAttached": false,
    "collapsed": false,
    "createdAt": "2022-09-19T09:28:05.066Z",
    "updatedAt": "2022-09-19T09:28:05.066Z",
    "sitemap_exclude": false,
    "audience": [
      
    ],
    "parent": null,
    "related": {
      "id": 1,
      "name": "About us",
      "slug": "/about-us",
      "createdAt": null,
      "updatedAt": "2022-06-12T21:05:06.497Z",
      "publishedAt": "2021-12-10T17:24:40.242Z",
      "locale": "de",
      "sitemap_exclude": null,
      "__contentType": "api::page.page",
      "navigationItemId": 29
    },
    "items": null
  },
  {
    "order": 1,
    "id": 30,
    "title": "Team",
    "type": "INTERNAL",
    "path": "team",
    "externalPath": null,
    "uiRouterKey": "team",
    "menuAttached": true,
    "collapsed": false,
    "createdAt": "2022-09-19T09:28:05.116Z",
    "updatedAt": "2022-09-19T09:55:54.945Z",
    "sitemap_exclude": false,
    "audience": [
      
    ],
    "parent": {
      "id": 29,
      "title": "About us",
      "type": "INTERNAL",
      "path": "about-us",
      "externalPath": null,
      "uiRouterKey": "about-us",
      "menuAttached": false,
      "order": 1,
      "collapsed": false,
      "createdAt": "2022-09-19T09:28:05.066Z",
      "updatedAt": "2022-09-19T09:28:05.066Z",
      "sitemap_exclude": false,
      "additionalFields": {
        
      }
    },
    "related": {
      "id": 18,
      "name": "About us -> Team",
      "slug": "/about-us/team",
      "createdAt": null,
      "updatedAt": "2022-06-12T21:05:24.676Z",
      "publishedAt": "2021-12-12T12:00:41.944Z",
      "locale": "de",
      "sitemap_exclude": null,
      "__contentType": "api::page.page",
      "navigationItemId": 30
    },
    "items": null
  },
  {
    "order": 2,
    "id": 31,
    "title": "Vision",
    "type": "INTERNAL",
    "path": "vision",
    "externalPath": null,
    "uiRouterKey": "vision",
    "menuAttached": false,
    "collapsed": false,
    "createdAt": "2022-09-19T09:28:05.117Z",
    "updatedAt": "2022-09-19T09:28:05.117Z",
    "sitemap_exclude": false,
    "audience": [
      
    ],
    "parent": {
      "id": 29,
      "title": "About us",
      "type": "INTERNAL",
      "path": "about-us",
      "externalPath": null,
      "uiRouterKey": "about-us",
      "menuAttached": false,
      "order": 1,
      "collapsed": false,
      "createdAt": "2022-09-19T09:28:05.066Z",
      "updatedAt": "2022-09-19T09:28:05.066Z",
      "sitemap_exclude": false,
      "additionalFields": {
        
      }
    },
    "related": {
      "id": 17,
      "name": "About us -> Vision",
      "slug": "/about-us/vision",
      "createdAt": null,
      "updatedAt": "2022-06-11T17:23:47.103Z",
      "publishedAt": "2021-12-12T11:57:21.215Z",
      "locale": "de",
      "sitemap_exclude": null,
      "__contentType": "api::page.page",
      "navigationItemId": 31
    },
    "items": null
  }
]
```

**Comment by ltsNotMike** [comment]
Added to the backlog. 

1. Add an option to disable the propagation of `menuAttached` property to nested elements. 
2. Propagation should be enabled by default, not to conflict with older versions.
3. User can edit the option both in the config file and on the Settings page.

**Comment by cyp3rius** [comment]
Released as part of version `2.2.2`.
